app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -84,4 +110,18 @@ def clear_session_data
     user_session.delete(unconfirmed_phone_session_key)
     user_session.delete(confirmation_code_session_key)
   end
+
+  def sms_enabled?
```

Comment:
Why not use dedicated classes to handle each delivery type? In general, concerns should be avoided. The one specific scenario where they are acceptable is to share code between several controllers, where all the controllers make use of all the shared code. Using concerns as a way to implement Inheritance is not a good pattern IMO.

In this case, I would recommend the [Strategy pattern](https://en.wikipedia.org/wiki/Strategy_pattern) over Inheritance because the separation makes things easier to understand, to maintain, to extend, and to reuse.

> According to the strategy pattern, the behaviors of a class should not be inherited. Instead they should be encapsulated using interfaces. As an example, consider a car class. Two possible functionalities for car are brake and accelerate.

In our case, the car is the OTP delivery, and there are two (or potentially more in the future) ways to deliver the code: sms and voice.
These different modes of delivery might not necessarily always be needed at the same time. For example, if we only wanted to use SMS during IdV, it would be a lot easier to use the Strategy pattern so that each object that needs to use a particular delivery mechanism can choose it when needed, without requiring significant code changes. 

> This allows better decoupling between the behavior and the class that uses the behavior.

Here are some related articles for Ruby:
https://robots.thoughtbot.com/back-to-basics-solid
https://robots.thoughtbot.com/decorators-compared-to-strategies-composites-and